### PR TITLE
Remove bootstrap and signal handlers

### PR DESCRIPTION
**Pull request by kevinburkeshyp**
App lifting and lowering should be handled entirely in user-land, where the
user can combine lowering with other shutdown tasks. Once we merge this change,
we'll need to update the API to handle SIGTERM and shut itself down. This is
a benefit, since our shutdown handlers will finally actually run.

Removes the `sails.config.bootstrap` option, which introduces unnecessary
callback complexity to Sails and is unused in our codebase.

Fixes https://github.com/balderdashy/sails/issues/3693.

### CODE REVIEW

File: lib/app/lift.js [hunk 1]
```diff
@@ -46,10 +38,9 @@ module.exports = function lift(configOverride, cb) {
       });
     }
 
+    sails.initialize();
```
Comment (kevinburkeshyp) [inline lib/app/lift.js:41]: This is now a synchronous function, we don't need to load it as part of an `async` chain.

**Comment by benbuckman** [comment]
Nice

**Comment by kevinburkeshyp** [comment]
@benbuckman is this a +1 or just a casual look?

**Comment by misterMuyiwa** [comment]
LGTM

**Comment by benbuckman** [comment]
> is this a +1 or just a casual look?

Sorry, it was just a casual look, I should have been clearer.

**Pull request merged**